update ds/block-array (#34)
OI-wiki/i18n: docs/ds/block-array.md

## Building block arrays

> 块状数组，即把一个数组分为几个块，块内信息整体保存，若查询时遇到两边不完整的块直接暴力查询。一般情况下，块的长度为 $O(\sqrt{n})$ 。详细分析可以阅读 2017 年国家集训队论文中徐明宽的《非常规大小分块算法初探》。

Reviewer: 中文没删

Author: 🌚

下面直接给出一种建立块状数组的代码。

> Block array is an array divided into several blocks, and the information within the block is stored as a whole. If you encounter incomplete blocks on both sides of the query, you can directly query brute-force. In general, the length of the block is $O(\sqrt{n})$ . For detailed analysis, you can read "A Preliminary Study on Unconventional Size Blocking Algorithms" by Xu Mingkuan in the 2017 National Training Team Paper.

Reviewer: ```suggestion
Block array is an array divided into several blocks, and the information within the block is stored as a whole. If you encounter incomplete blocks on both sides while handling queries, you can directly query with brute-force. In general, the length of the block is $O(\sqrt{n})$ . For detailed analysis, you can read "A Preliminary Study on Unconventional Size Blocking Algorithms" by Xu Mingkuan in the 2017 National Training Team Papers.
```

}
```

> `st[i] ed[i]` is the start and end of the block, and `size[i]` is the size of the block.

Reviewer: ```suggestion
`st[i]` here `ed[i]`  are the start and end point of the block, and `size[i]` is the size of the block.
```

Reviewer: 🌚 我写的个啥玩意 
`st[i]` and `ed[i]` here are the start and end points

## Save and modify information in the block

### Sample problem 1： [Master's Magic](https://www.luogu.com.cn/problem/P2801) (original link in Chinese)

> We need to query the number of numbers greater than or equal to a number in a block, so we need a `t` array to sort the blocks. For the modification of the whole block, it is saved in a way similar to the permanent mark. Time complexity is $O(n\sqrt{n}\log n)$

Reviewer: 标记永久化没体现出`化`诶

Author: ...making the mark permanent ? 🤔

}
```

### Sample problem 2: Cold Night Ark

Two operations:

> 1. Every number in the interval $[x,y]$ becomes $z$

Reviewer: change ... into

}
```

## Practices

> > Following problems are written in Chinese.

Reviewer: ```suggestion
> The Following problems are written in Chinese.
```
